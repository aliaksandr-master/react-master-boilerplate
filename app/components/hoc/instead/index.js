import instead from './instead';



export default instead;
