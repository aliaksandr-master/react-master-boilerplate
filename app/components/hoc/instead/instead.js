import isString from 'lodash/isString';
import { branch, renderComponent, defaultProps } from 'recompose';
import WaitingText, { THEME } from '../../ui/WaitingText';



const BlockWaitingText = defaultProps({ theme: THEME.BLOCK, height: 300 })(WaitingText);


export default (condition, Component = () => null) => {
  if (isString(condition)) {
    const propName = condition;

    condition = (props) => Boolean(props[propName]);
  }

  if (Component === true) {
    Component = BlockWaitingText;
  }

  return branch(
    condition,
    renderComponent(Component)
  );
};
