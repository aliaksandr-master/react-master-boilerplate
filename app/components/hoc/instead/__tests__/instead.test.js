/* eslint-env browser, jest */
import React from 'react';
import { mapProps, compose } from 'recompose';
import renderer from 'react-test-renderer';
import instead from '../index';
import Test from '../../../test/Test';
import div from '../../../../lib/root/index';



const ComponentPos = compose(
  instead(() => true, () => <div />)
)(() => <noscript />);

const ComponentNeg = compose(
  instead(() => false, () => <div />)
)(() => <noscript />);

const ComponentDef = compose(
  instead('hello', () => <div />)
)(() => <noscript />);

describe('rendering', () => {
  it('should render something1', () => {
    const component = renderer.create(
      <Test>
        <ComponentPos />
      </Test>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should render something2', () => {
    const component = renderer.create(
      <Test>
        <ComponentNeg />
      </Test>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should render something3', () => {
    const component = renderer.create(
      <Test>
        <ComponentDef />
      </Test>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
