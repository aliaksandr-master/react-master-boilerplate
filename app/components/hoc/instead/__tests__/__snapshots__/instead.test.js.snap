// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render something1 1`] = `<div />`;

exports[`rendering should render something2 1`] = `<noscript />`;

exports[`rendering should render something3 1`] = `<noscript />`;
